### PR TITLE
0.14 RC blog post

### PR DESCRIPTION
**Pull request by spicyj**
Apparently we did a lot.

To curious onlookers: Please no tweeting about this yet; we're not quite done. :) We'll try to cut the release soon.

Preview:

### CODE REVIEW

**Comment by gaearon** [comment]
Amazing job @spicyj @cpojer @zpao @sebmarkbage & co, this is going to be my favorite release so far!

**Comment by mridgway** [comment]
The deprecation of `this.getDOMNode()` should probably refer to `ReactDOM.getDOMNode(this)` instead of `React.getDOMNode(this)` since it is deprecated in React package.

Can't wait for this! :+1:

**Comment by spicyj** [comment]
Thank you, good catch.

**Comment by knowbody** [comment]
Great work! Huge release, looking forward to it

**Comment by korczis** [comment]
:+1:

**Comment by alexeyraspopov** [comment]
How stateless function components supposed to work? Or should I still use classes to wrap these stateless functions?

**Comment by bloodyowl** [comment]
great job. can't wait for this release!

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        return <div>Hello World</div>;
+      }
+    });
+
+    ReactDOM.render(<MyComponent />, node);
+    ```
+
+    We've published the [automated codemod script](https://github.com/facebook/react/blob/master/packages/react-codemod/README.md) we used at Facebook to help you with this transition.
+
+    The addons have moved to separate packages as well: `react-addons-clone-with-props`, `react-addons-create-fragment`, `react-addons-css-transition-group`, `react-addons-linked-state-mixin`, `react-addons-pure-render-mixin`, `react-addons-shallow-compare`, `react-addons-transition-group`, and `react-addons-update`, plus `ReactDOM.unstable_batchedUpdates` in `react-dom`.
```
Comment (koba04) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: @spicyj `react-addons-perf` and `react-addons-test-utils` too?

**Comment by yaycmyk** [comment]
@alexeyraspopov It was explained pretty clearly in the post, IMO. Instead of having to `createClass` or extend `React.Component`, you simply feed a function with props as the primary argument and it takes the static output from it to render.

**Comment by alexeyraspopov** [comment]
@yaycmyk I've tried to do it with 0.14-beta3 and it doesn't work.

**Comment by yaycmyk** [comment]
@alexeyraspopov AFAIK it's not actually merged yet - https://github.com/facebook/react/pull/3995

**Comment by spicyj** [comment]
@yaycmyk It's not in beta3. Please wait for the RC.

**Comment by alexeyraspopov** [comment]
@spicyj but what about #4587?

**Comment by spicyj** [comment]
@alexeyraspopov What did I _just_ say?

**Comment by spicyj** [comment]
@alexeyraspopov That landed after beta3 was cut. It'll be in the RC that goes out sometime soon.

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+We've also published these files in the `react` and `react-dom` packages on bower.
+
+## Changelog
+
+### Major changes
+
+- **Two Packages**
+
+    As we look at packages like [react-native](https://github.com/facebook/react-native), [react-art](https://github.com/reactjs/react-art), [react-canvas](https://github.com/Flipboard/react-canvas), and [react-three](https://github.com/Izzimach/react-three), it's become clear that the beauty and essence of React has nothing to do with browsers or the DOM.
```
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: What's with indenting paragraphs?

Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: s/it's/it has/

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  Dev build with warnings: <https://fb.me/react-dom-0.14.0-rc1.js>  
+  Minified build for production: <https://fb.me/react-dom-0.14.0-rc1.min.js>  
+
+We've also published these files in the `react` and `react-dom` packages on bower.
+
+## Changelog
+
+### Major changes
+
+- **Two Packages**
```
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: Do we want to use `h4`?

Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: Looks the same. I'm fine with either if you have a preference.

File: docs/_posts/2015-09-04-react-v0.14-rc1.md (hunk 2 above)
Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: Makes it part of the same bullet point.

Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: ack

File: docs/_posts/2015-09-04-react-v0.14-rc1.md (hunk 1 above)
Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: Yes, thanks.

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        {getFish(species)}
+      </Tank>
+    );
+    ```
+
+    This pattern is designed to encourage the creation of these simple components that should comprise large portions of your apps. In the future, we'll also be able to make performance optimizations specific to these components by avoiding unnecessary checks and memory allocations.
+
+- **Deprecation of react-tools**
+
+    The `react-tools` package and `JSXTransformer.js` browser file have been deprecated. You can continue using version `0.13.3` of both, but we no longer support them and recommend migrating to [Babel](http://babeljs.io/), which has built-in support for React and JSX.
```
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: > have been deprecated

Maybe link to the deprecation post?

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+
+### Breaking changes
+
+As always, we have a few breaking changes in this release. Whenever we make large changes, we warn for at least one release so you have time to update your code. Our codebase has over 15,000 React components, so we always try to minimize the pain of breaking changes.
+
+These three breaking changes had a warning in 0.13, so you shouldn't have to do anything if your code is already free of warnings:
+
+- The `props` object is now frozen, so mutating props after creating a component element is no longer supported. In most cases, [`React.cloneElement`](http://facebook.github.io/react/docs/top-level-api.html#react.cloneelement) should be used instead. This change makes your components easier to reason about and enables the compiler optimizations mentioned above.
+- Plain objects are no longer supported as React children; arrays should be used instead. You can use the [createFragment](/react/docs/create-fragment.html) helper to migrate, which now returns an array.
```
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: \`createFragment\` for better formatting

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    **Constant hoisting for React elements:** The `optimisation.react.constantElements` transform hoists element creation to the top level for subtrees that are fully static, which reduces calls to `React.createElement` and the resulting allocations. More importantly, it tells React that the subtree hasn't changed so React can completely skip it when reconciling.
+
+
+### Breaking changes
+
+As always, we have a few breaking changes in this release. Whenever we make large changes, we warn for at least one release so you have time to update your code. Our codebase has over 15,000 React components, so we always try to minimize the pain of breaking changes.
+
+These three breaking changes had a warning in 0.13, so you shouldn't have to do anything if your code is already free of warnings:
+
+- The `props` object is now frozen, so mutating props after creating a component element is no longer supported. In most cases, [`React.cloneElement`](http://facebook.github.io/react/docs/top-level-api.html#react.cloneelement) should be used instead. This change makes your components easier to reason about and enables the compiler optimizations mentioned above.
```
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: relative link?

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+- Plain objects are no longer supported as React children; arrays should be used instead. You can use the [createFragment](/react/docs/create-fragment.html) helper to migrate, which now returns an array.
+- Add-Ons: `classSet` has been removed. Use [classnames](https://github.com/JedWatson/classnames) instead.
+
+These two changes did not have warnings in 0.13 but should be easy to find and clean up:
```
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: `classSet` did and I thought we did for using `createFragment` as well…

Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: This refers to the following list. Is that unclear?

File: docs/_posts/2015-09-04-react-v0.14-rc1.md (hunk 6 above)
Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: ack (twice in this file)

File: docs/_posts/2015-09-04-react-v0.14-rc1.md (hunk 5 above)
Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: ack

File: docs/_posts/2015-09-04-react-v0.14-rc1.md (hunk 7 above)
Comment (spicyj) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: changed "These" to "And these"

File: docs/_posts/2015-09-04-react-v0.14-rc1.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+- React DOM now supports these SVG attributes, which render into namespaced attributes: `xlinkActuate`, `xlinkArcrole`, `xlinkHref`, `xlinkRole`, `xlinkShow`, `xlinkTitle`, `xlinkType`, `xmlBase`, `xmlLang`, `xmlSpace`.
+- The `image` SVG tag is now supported by React DOM.
+- In React DOM, arbitrary attributes are supported on custom elements (those with a hyphen in the tag name or an `is="..."` attribute).
+- React DOM now supports these media events on `audio` and `video` tags: `onAbort`, `onCanPlay`, `onCanPlayThrough`, `onDurationChange`, `onEmptied`, `onEncrypted`, `onEnded`, `onError`, `onLoadedData`, `onLoadedMetadata`, `onLoadStart`, `onPause`, `onPlay`, `onPlaying`, `onProgress`, `onRateChange`, `onSeeked`, `onSeeking`, `onStalled`, `onSuspend`, `onTimeUpdate`, `onVolumeChange`, `onWaiting`.
+- A [shallowCompare](https://github.com/facebook/react/pull/3355) addon has been added as a migration path for `PureRenderMixin` in ES6 classes.
```
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: More code formatting

File: docs/_posts/2015-09-04-react-v0.14-rc1.md (hunk 7 above)
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: Oh I misread. It's quite clear on a 2nd read.

File: docs/_posts/2015-09-04-react-v0.14-rc1.md (hunk 2 above)
Comment (zpao) [inline docs/_posts/2015-09-04-react-v0.14-rc1.md]: Oh that's a giant list… that was hard to see in the diff. I guess that's ok (and then you can ignore the `h4` comment).

**Comment by spicyj** [comment]
(marking accepted per 1:1 chat)

**Pull request merged**